### ISSUE TITLE
[Tune] PBT scheduler ignoring some of ``hyperparam_mutations`` after restore

### ISSUE THREAD

**Issue by davidhozic** [ray-project/ray#54755]
### What happened + What you expected to happen

When using ``PopulationBasedTraining`` and restoring from a checkpoint, some of the original ``hyperparameter_mutations`` seem to be ignored, resulting in resampling returning the same value.

In the following example, ``clip_param`` seems to be set to 0. Thus, no gradients are actually being optimized, apart for the specific case when the ratio in the PPO's objective is below 1. 

```
lr : 0.0006183230705195503 --- (* 0.8) --> 0.0004946584564156402
num_sgd_iter : 1 --- (* 0.8) --> 1
entropy_coeff : 0.0005671363125642761 --- (resample) --> 0.0009495159664290651
clip_param : 0 --- (resample) --> 0
lambda : 0.9683199949528589 --- (* 0.8) --> 0.7746559959622872
```

The parameter of hyperparameter mutations is as follows:
```
hyperparam_mutations={  
    "lr": tune.uniform(1e-5, 1e-3),
    "clip_param": tune.uniform(0.005, 0.5),
    "entropy_coeff": tune.uniform(0.0, 0.002),
    "lambda": tune.uniform(0.5, 1.0),
    "num_sgd_iter": tune.randint(3, 15),
},
```

### Versions / Dependencies

- Ray 2.10.0
- Python 3.11.11

### Reproduction script

```py
PATH = "/root/ray_results/PPOWithCurriculum_2025-07-15_14-44-17"

os.environ["TUNE_DISABLE_AUTO_CALLBACK_LOGGERS"] = "1"  # Don't create result.json and progress.csv
if PATH:
    tuner = tune.Tuner.restore(PATH, PPOWithCurriculum, resume_errored=True, resume_unfinished=True)
else:
    tuner = tune.Tuner(
        "PPO",
        param_space=config,
        run_config=train.RunConfig(
            checkpoint_config=train.CheckpointConfig(10),
            callbacks=[TBXReducedLoggerCallback()],
            # stop={"timesteps_total": 20e6}
        ),
        tune_config=tune.TuneConfig(
            metric="episode_reward_mean",
            mode="max",
            trial_dirname_creator=lambda trial: str(next(cnt)),
            num_samples=20,
            scheduler=PopulationBasedTraining(
                # perturbation_interval=10 * 60,
                perturbation_interval=100,
                time_attr="training_iteration",
                hyperparam_mutations={  
                    "lr": tune.uniform(1e-5, 1e-3),
                    "clip_param": tune.uniform(0.005, 0.5),
                    "entropy_coeff": tune.uniform(0.0, 0.002),
                    "lambda": tune.uniform(0.5, 1.0),
                    "num_sgd_iter": tune.randint(3, 15),
                },
                custom_explore_fn=check_parameters,
            ),
        ),
    )

results = tuner.fit()
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by davidhozic** [comment]
I was able to somewhat overcome this by manually setting the scheduler after restore to ``tuner._local_tuner._tune_config.scheduler = scheduler``, where I updated the explore function to clamp between 0.005 and 1, where originally only clamping to 1 was done.

**Comment by simonsays1980** [comment]
@davidhozic Working with PBT or PB2 schedulers right now with RLlib appears to be somehow a bit off. I noticed this myself when using PB2 in an experiment. 

Dumb question: Do you use some `tune` distributions for these hyperparameters in the RLlib config?

**Comment by davidhozic** [comment]
> [@davidhozic](https://github.com/davidhozic) Working with PBT or PB2 schedulers right now with RLlib appears to be somehow a bit off. I noticed this myself when using PB2 in an experiment.
> 
> Dumb question: Do you use some `tune` distributions for these hyperparameters in the RLlib config?

Yes, some parameters are randomized at beginning using tune's distributions.

```
lr=tune.uniform(6e-05, 5e-05),
clip_param=tune.uniform(0.3, 0.4),
num_sgd_iter=tune.randint(8, 12),
lambda_=tune.uniform(0.90, 1.0),
entropy_coeff=tune.uniform(0.000, 0.005),
```

I also have tune's distributions in `hyperparameter_mutations` of PBT (as shown above).

**Comment by davidhozic** [comment]
> > [@davidhozic](https://github.com/davidhozic) Working with PBT or PB2 schedulers right now with RLlib appears to be somehow a bit off. I noticed this myself when using PB2 in an experiment.
> > Dumb question: Do you use some `tune` distributions for these hyperparameters in the RLlib config?
> 
> Yes, some parameters are randomized at beginning using tune's distributions.
> 
> ```
> lr=tune.uniform(6e-05, 5e-05),
> clip_param=tune.uniform(0.3, 0.4),
> num_sgd_iter=tune.randint(8, 12),
> lambda_=tune.uniform(0.90, 1.0),
> entropy_coeff=tune.uniform(0.000, 0.005),
> ```
> 
> I also have tune's distributions in `hyperparameter_mutations` of PBT (as shown above).

Although, I don't think that's the issue. It seems to always happen when restoring trials. I was able to reduce it by simply not allowing trials to pause unless they were the weak trials which were replaced by good trials. Now it only happens on the trials that were on the bottom of the quantile. Still not ideal, but at least the good trials don't get nerfed.

**Comment by davidhozic** [comment]
Update: I've just finished training with PBT where the training graphs look like the models were 90% successful. If I restart the training they all drop to like 40 %, same with evaluation. It's definitely a checkpointing thing...

**Comment by lukasheuer** [comment]
Is there any update on this? 
I had a similar issue when copying checkpoints over while running pbt.
